src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -241,6 +248,52 @@ public void initializeFromItineraryAndQueryParams() throws IllegalArgumentExcept
         }
     }
 
+    /**
+     * Updates the mode in the OTP query parameter based on the modes from the itinerary.
+     */
+    private void updateModeInQueryParams() throws URISyntaxException {
+        Set<String> actualModes = itinerary.legs.stream()
```

Comment:
Can this be renamed `legModes` to improve readability?

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/104/commits/1354bae3d3917783f408d667ef115e7435db5cf5.